File: examples/go/run/event.go
package main

import (
	"context"

	v1_workflows "github.com/hatchet-dev/hatchet/examples/go/workflows"
	v1 "github.com/hatchet-dev/hatchet/pkg/v1"
	"github.com/joho/godotenv"
)

func event() {
	err := godotenv.Load()
	if err != nil {
		panic(err)
	}

	hatchet, err := v1.NewHatchetClient()

	if err != nil {
		panic(err)
	}
	// > Pushing an Event
	err = hatchet.Events().Push(
		context.Background(),
		"simple-event:create",
		v1_workflows.SimpleInput{
			Message: "Hello, World!",
		},
	)
	

	if err != nil {
		panic(err)
	}
}

File: examples/typescript/on_event copy/event.ts
import { hatchet } from '../hatchet-client';
import { Input } from './workflow';

async function main() {
  // ❓ Pushing an Event
  const res = await hatchet.event.push<Input>('simple-event:create', {
    Message: 'hello',
  });

  console.log(res.eventId);
}

if (require.main === module) {
  main();
}
